src/store/auth/actions.js

```diff
@@ -40,3 +43,18 @@ export const logout = () => (dispatch) => {
   dispatch(removeUserInfo())
   // })
 }
+
+
+export const refreshToken = () => (dispatch, getState) => {
+  const { _token, refreshToken, _id, name } = getState().user;
+
+  getNewToken(refreshToken, _token, _id, name)
+    .then(res => {

```

Comment:
I think, **then** should almost always be with **catch**. It extremely important in mobile app, because if user sees an error screen he is very likely delete application and it will be very **bad user experience**. This means that in mobile applications, there should **never** be an **error screen**, there should be a correct error output on the UI